src/utils/env/index.ts

```diff
@@ -1,65 +1,117 @@
 import type { NetlifyAPI } from 'netlify'
 
-import { $TSFixMe } from '../../commands/types.js'
 import { logAndThrowError } from '../command-helpers.js'
 import type { SiteInfo, EnvironmentVariableSource } from '../../utils/types.js'
 
-export const AVAILABLE_CONTEXTS = ['all', 'production', 'deploy-preview', 'branch-deploy', 'dev']
-export const AVAILABLE_SCOPES = ['builds', 'functions', 'runtime', 'post_processing']
-
-type EnvironmentVariableContext = 'all' | 'production' | 'deploy-preview' | 'branch-deploy' | 'dev'
-
-type EnvironmentVariableScope = 'builds' | 'functions' | 'runtime' | 'post_processing'
+/**
+ * Supported values for the user-provided env `context` option.
+ * These all match possible `context` values returned by the Envelope API.
+ * Note that a user may also specify a branch name with the special `branch:my-branch-name` format.
+ */
+export const SUPPORTED_CONTEXTS = ['all', 'production', 'deploy-preview', 'branch-deploy', 'dev'] as const
+/**
+ * Additional aliases for the user-provided env `context` option.
+ */
+const SUPPORTED_CONTEXT_ALIASES = {
+  dp: 'deploy-preview',
+  prod: 'production',
+}
+/**
+ * Supported values for the user-provided env `scope` option.
+ * These exactly match possible `scope` values returned by the Envelope API.
+ * Note that `any` is also supported.
+ */
+export const ALL_ENVELOPE_SCOPES = ['builds', 'functions', 'runtime', 'post_processing'] as const
 
-type EnvironmentVariableValue = {
-  context: EnvironmentVariableContext
+// TODO(serhalp) Netlify API is incorrect - the returned scope is `post_processing`, not `post-processing`
+type EnvelopeEnvVarScope =
+  | Exclude<NonNullable<Awaited<ReturnType<NetlifyAPI['getEnvVars']>>[number]['scopes']>[number], 'post-processing'>
+  | 'post_processing'
+type EnvelopeEnvVar = Awaited<ReturnType<NetlifyAPI['getEnvVars']>>[number] & {
+  scopes: EnvelopeEnvVarScope[]
+}
+type EnvelopeEnvVarContext = NonNullable<EnvelopeEnvVar['values']>[number]['context']
```

Comment:
Do you mean for `undefined` to be part of this union? The intent here is _really_ unclear to me because of all the type magic.

Reply:
Good catch, thanks. No, that was not intentional.